vulnfeeds/git/versions.go

```diff
@@ -16,25 +16,36 @@ package git
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 
 	"golang.org/x/exp/maps"
 
 	"github.com/google/osv/vulnfeeds/cves"
 )
 
-// Take an already normalized version, repo and the pre-normalized mapping of tags to commits and do fuzzy matching on the version, returning a GitCommit and a bool if successful.
+// Take an already normalized version, repo and the mapping of repo tags
+// normalized tags and commits and do fuzzy matching version, returning a
+// GitCommit and a bool if successful.
 func fuzzyVersionToCommit(normalizedVersion string, repo string, commitType cves.CommitType, normalizedTags map[string]NormalizedTag) (ac cves.AffectedCommit, b bool) {
-	candidateTags := []string{}
+	candidateTags := []string{} // the subset of normalizedTags tags that might be appropriate to use as a fuzzy match for normalizedVersion.
+	// Keep in sync with the regex in cves.NormalizeVersion()
+	var validVersionText = regexp.MustCompile(`(?i)(?:rc|alpha|beta|preview)\d*`)
+
 	for _, k := range maps.Keys(normalizedTags) {
-		if strings.HasPrefix(k, normalizedVersion) {
+		// "1.8.0-RC0" shouldn't be considered a fuzzy match for "1.8.0"
+		if strings.HasPrefix(k, normalizedVersion) && !validVersionText.MatchString(k) {
```

Comment:
Will this exclude legitimate rc/beta/alpha etc matches? i.e. ifr both the normalized tag and the `normalizedVersion` have -rc1 or something similar?

Reply:
Maybe the check here should be: either both the normalized tag and version have a pre-release string tag (i.e. matches `validVersionText`), or none do?

Reply:
Yeah I think you're right. I also found a case sensitivity issue that was mishandling exact-match checks with versions with text in them. All addressed in 193a37b